File: cpp/hot100/31.K 个一组翻转链表.cpp
class Solution {
public:
    ListNode* reverseKGroup(ListNode* head, int k) {
        ListNode *dummy=new ListNode(0);
        dummy->next=head;
        ListNode *pre=dummy;
        ListNode *lat=head;
        while(lat){
            ListNode *check=lat;
            for(int i=0;i<k;i++){
                if(!check) return dummy->next;
                check=check->next;
            }
            ListNode *tail=lat;
            for(int i=0;i<k-1;i++){
                ListNode *temp=lat->next;
                lat->next=temp->next;
                temp->next=pre->next;
                pre->next=temp;
            }
            pre=tail;  
            lat=pre->next;
        }
        return dummy->next;
    }
};

File: python/hot100/31.K 个一组翻转链表.py
class Solution:
    def reverseKGroup(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        dummy=ListNode(0,head)
        pre,cur=dummy,head
        while(cur):
            check=cur
            for i in range(0,k):
                if check == None:
                    return dummy.next
                check=check.next
            tail=cur
            for i in range(0,k-1):
                temp=cur.next
                tail.next=temp.next
                temp.next=pre.next
                pre.next=temp
            pre=tail
            cur=pre.next
        return dummy.next